core/constraints/constraints.go

```diff
@@ -330,18 +330,28 @@ func Parse(args ...string) (Value, error) {
 func ParseWithAliases(args ...string) (cons Value, aliases map[string]string, err error) {
 	aliases = make(map[string]string)
 	for _, arg := range args {
+		name, val := "", ""
 		raws := strings.Split(strings.TrimSpace(arg), " ")
 		for _, raw := range raws {
 			if raw == "" {
 				continue
 			}
-			name, val, err := splitRaw(raw)
+			current_name, current_val, err := splitRaw(raw)
 			if err != nil {
 				return Value{}, nil, errors.Trace(err)
 			}
-			if canonical, ok := rawAliases[name]; ok {
-				aliases[name] = canonical
-				name = canonical
+			if current_name == "" && name == "" {
+				return Value{}, nil, errors.Errorf("malformed constraint %q", current_val)
+			}
+			if current_name != "" {
+				name = current_name
+				val = current_val
+				if canonical, ok := rawAliases[current_name]; ok {
+					aliases[current_name] = canonical
+					current_name = canonical
```

Comment:
@SimonRichardson From a second look, I believe this is wrong and should instead read:

```
if canonical, ok := rawAliases[name]; ok {
	aliases[name] = canonical
	name = canonical
}
```

Note that L351 triggers an ineffassign warning; not sure why it didn't complain on develop.